### PR TITLE
Optics

### PR DESCRIPTION
**Pull request by rpiaggio**

### CODE REVIEW

File: model/src/main/scala/clue/data/InputOptics.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package clue.data
+
+import monocle.PPrism
+import monocle.Prism
+
+trait InputOptics {
+  final def pAssign[A, B]: PPrism[Input[A], Input[B], A, B] =
+    PPrism[Input[A], Input[B], A, B](_.toOption.map(Right(_)).getOrElse(Left(Ignore)))(Assign.apply)
+
+  final def assign[A]: Prism[Input[A], A] =
```
Comment (cquiroz) [inline model/src/main/scala/clue/data/InputOptics.scala:13]: Should these have tests?

File: model/src/main/scala/clue/data/syntax/package.scala [hunk 2]
```diff
@@ -12,4 +16,17 @@ package object syntax {
     def orIgnore: Input[A]   = Input.orIgnore(a)
     def orUnassign: Input[A] = Input.orUnassign(a)
   }
+
+  implicit final class LensToLensOps[S, T, A, B](private val lens: PLens[S, T, A, B])
```
Comment (cquiroz) [inline model/src/main/scala/clue/data/syntax/package.scala:20]: Are we using  this method instead of `extension` to remain cross compilable to scala 2?

**Review by cquiroz — APPROVED** [review]
Thanks it looks great

File: model/src/main/scala/clue/data/InputOptics.scala (hunk 1 above)
Comment (rpiaggio) [reply model/src/main/scala/clue/data/InputOptics.scala:13]: Yup, that's why it's still a draft.

File: model/src/main/scala/clue/data/syntax/package.scala (hunk 2 above)
Comment (rpiaggio) [reply model/src/main/scala/clue/data/syntax/package.scala:20]: Yes

**Pull request merged**